src/cli.ts

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env node
+
+import { program } from "commander";
+import fs from "fs";
+import chalk from "chalk";
+
+import { getFilesFromInput } from "./getFilesFromInput";
+import { analyzeFile } from "./analyzeFile";
+import { printDiagnostics } from "./printDiagnostics";
+import { Stats, getStats } from "./getStats";
+
+type PackageJSON = {
+  version: `${number}.${number}.${number}`;
+};
+
+type Plugin = {
+  name: string;
+  [key: string]: unknown;
+};
+
+const { version }: PackageJSON = require("../package.json");
+
+program
+  .description("An application to lint your TODO comments")
+  .version(version)
+  .argument("<file|dir|glob...>", "Files to check for comments")
+  .action(async (input) => {
+    const [error, files] = await getFilesFromInput(input);
+
+    if (error) {
+      console.error(error.message);
+      process.exit(1);
+    }
+
+    if (!files) {
+      console.error("No files found.");
+      process.exit(1);
+    }
+
+    const stats: Stats = {
+      errors: 0,
+      warnings: 0,
+    };
+
+    const currentDirectory = process.cwd();
+
+    // Read project package.json
+    const packageJsonPath = `${currentDirectory}/package.json`;
+
+    if (!fs.existsSync(packageJsonPath)) {
+      console.error(
+        `"package.json" not found. Make sure to run this command from the root folder of your project.`
+      );
+      process.exit(1);
+    }
+
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf8"));
+
+    // Read project tsconfig.json
+    const tsconfigJsonPath = `${currentDirectory}/tsconfig.json`;
+
+    if (!fs.existsSync(tsconfigJsonPath)) {
+      console.error(
+        `"tsconfig.json" not found. Make sure to run this command from the root folder of a TypeScript project.`
+      );
+      process.exit(1);
+    }
+
+    const tsconfigJson = JSON.parse(fs.readFileSync(tsconfigJsonPath, "utf8"));
+    const todoOrDiePlugin = tsconfigJson?.compilerOptions?.plugins?.find(
+      (plugin: Plugin) => plugin.name === "typescript-todo-or-die-plugin"
+    );
+
+    const options = todoOrDiePlugin?.options;
+    const additionalKeywords = todoOrDiePlugin?.additionalKeywords || [];
+
+    files.forEach((file) => {
```

Comment:
Maybe not feasible but just thinking about if could make the diagnose of files async so in theory could speed up the process by parallelization...maybe even use a stream..

Reply:
Good point! We could do that. We should make sure that when logging the output to the command line the files should always be in the same order. So maybe wait with printing until all files are analyzed. What do you think? Also I'll create a new issue for this instead of fixing it in this one. Is that okay?

Reply:
Yes, a new issue is the way to go, we probably need a bit of planning. I think that if we use a streaming lib like RxJS we can so some async stuff and wait an all to finish and then do some more transformations (ex printing)